tests/test_classification.py

```diff
@@ -47,157 +47,266 @@
+            dir_path.mkdir(exist_ok=True)
+
+    @classmethod
+    def tearDownClass(cls):
+        for path in [
+            train_dir / "train_single_file.jsonlines",
+            train_dir / "pos_label_train.jsonlines",
+            test_dir / "test_single_file.jsonlines",
+            test_dir / "test_single_file_subset.jsonlines",
+            output_dir / "rare_class_predictions.tsv",
+            output_dir / "float_class_predictions.tsv",
+        ]:
+            unlink(path)
+
+        for path in chain(
+            output_dir.glob("pos_label_predict*"),
+            output_dir.glob("train_test_single_file*"),
+            output_dir.glob("clf_metric_value_*"),
+            output_dir.glob("pos_label_via_config*"),
+            output_dir.glob("clf_metrics_objective_overlap*"),
+            output_dir.glob("test_multinomialnb_loading*"),
+            output_dir.glob("test_predict_return_and_write_predictions*"),
+        ):
+            unlink(path)
+
+        for config_file in chain(
+            [
+                config_dir / cfgname
+                for cfgname in ["test_single_file.cfg", "test_single_file_saved_subset.cfg"]
+            ],
+            config_dir.glob("test_metric_values_for_classification_*.cfg"),
+            config_dir.glob("test_fancy_pos_label*.cfg"),
+            config_dir.glob("test_fancy_metrics_objective_overlap*.cfg"),
+        ):
+            unlink(config_file)
+
+        for file_name in ["metric_values_test.jsonlines", "metric_values_train.jsonlines"]:
+            unlink(train_dir / file_name)
+
+    def test_contiguous_int_or_float_labels(self):
+        """Test that we can accurately detect contiguous int/float labels."""
+        self.assertEqual(contiguous_ints_or_floats([1, 2, 3, 4]), True)
+        self.assertEqual(contiguous_ints_or_floats([0, 1]), True)
+        self.assertEqual(contiguous_ints_or_floats([1.0, 2.0]), True)
+        self.assertEqual(contiguous_ints_or_floats([0, 1.0]), True)
+        self.assertEqual(contiguous_ints_or_floats([-2, -1, 0, 1, 2]), True)
+        self.assertEqual(contiguous_ints_or_floats([1.0, 2.0, 3.0, 4.0]), True)
+        self.assertEqual(contiguous_ints_or_floats([4, 5, 6]), True)
+        self.assertEqual(contiguous_ints_or_floats([1, 2, 3, 4.0]), True)
+        self.assertEqual(contiguous_ints_or_floats([-1, 1]), False)
+        self.assertEqual(contiguous_ints_or_floats([2, 4, 6]), False)
+        self.assertEqual(contiguous_ints_or_floats([3, 6, 11]), False)
+        self.assertEqual(contiguous_ints_or_floats([-2.0, -1.0, 1.0, 2.0]), False)
+        self.assertEqual(contiguous_ints_or_floats([1.0, 1.1, 1.2]), False)
+        self.assertRaises(TypeError, contiguous_ints_or_floats, ["a", "b", "c"])
+        self.assertRaises(TypeError, contiguous_ints_or_floats, np.array([1, 2, 3, "a"]))
+        self.assertRaises(ValueError, contiguous_ints_or_floats, [])
+        self.assertRaises(ValueError, contiguous_ints_or_floats, np.array([]))
```

Comment:
Not critical, but I'd prefer to see separate tests for error handling.

Reply:
Care to elaborate?